building-and-deploying-a-fastapi-app.md

```diff
@@ -0,0 +1,541 @@
+```
+
+With the installation complete, define a base route in `app/api.py`:
+
+```py
+from fastapi import FastAPI
+
+app = FastAPI()
+
+@app.get("/", tags=["Home"])
+def get_root() -> dict:
+    returrn {
+        "message": "Welcome to the okteto's app."
+    }
+```
+
+You started by importing the FastAPI class from the `fastapi` package in the code block above. Then you created an instance of the class in the variable `app`. 
+
+Next, you defined a `GET` route on __"/"__ which is handled by the `get_root()` function.
+
+
+In the `main.py` file, define an entry point for running the application:
+
+```py
+import uvicorn
+
+if __name__ == "__main__":
+    uvicorn.run("app.api:app", host="0.0.0.0", port=8080, reload=True)
+```
+
+In the code block above, you imported the uvicorn package itself. Under the initializer block, you invoked the `fun` method, which takes the location of FastAPI's instance, the host, port, and the reload boolean value.
+
+Next, start the application:
+
+```
+python main.py
+```
+
+
+You should get a response like this:
+
+```
+INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
+INFO:     Started reloader process [24513] using statreload
+INFO:     Started server process [24515]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+
+```
+
+Navigate to http://localhost:8080 in your browser. You should see:
+
+```
+{
+    "message": "Welcome to the okteto's app."
+}
+```
+
+## Routes
+
+You'll be building a recipe application.
+
+Before you begin writing the routes, define the model schema for the application:
+
+In `app/model.py`, write the following:
+
+```
+from pydantic import BaseModel, Field
+from typing import Optional, List
+
+
+class RecipeSchema(BaseModel):
+    id: Optional[int]
+    name: str = Field(...)
+    ingredients: List[str] = Field(...)
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "name": "Donuts",
+                "ingredients": ["Flour", "Milk", "Sugar", "Vegetable Oil"]
+            }
+        }
+
+
+class UpdateRecipeSchema(BaseModel):
+    name: Optional[str]
+    ingredients: Optional[List[str]]
+
+    class Config:
+        schema_extra = {
+            "example": {
+                "name": "Buns",
+                "ingredients": ["Flour", "Milk", "Sugar", "Vegetable Oil"]
+            }
+        }
+
+
+```
+
+In the block of code above, you defined how the recipe would be represented, sent, and retrieved in your in-app database. Each recipe will have an ID automatically generated once a POST request is sent. Each recipe will also comprise of a name and an array of ingredients. If any other type of data not listed in the schema is sent in the request body, an error will be returned.
+
+The `RecipeSchema` has a subclass `Config`. The subclass contains an object variable, `schema_extra`, which includes a key `example` used as the mock data in the interactive documentation.
```

Comment:
How would this be used in "prod"? It's not clear to me how to go from the mock data to a real data source.

Reply:
I'm not sure I understand what you mean. I'll explain what I mean by the statement above: in production, data will be sent according to the way recipes are represented in the schema. The mock data's work is to show an example of how the data is represented and how it should be sent in a request body.

In essence, the real data source should mimic the mock data.

Reply:
Makes sense. Could you briefly mention that, for the purpose of this example, we are using an in-memory database, but that for realistic scenarios the user should use a separate DB?

Reply:
Sure thing, I did mention that somewhere in the article I believe. I'll check and mention it properly.